### PR TITLE
Nodetool access to the Modules

### PR DESCRIPTION
**Pull request by sylvestor88**
The devs need can invoke the nodetool method with following parameters:
  int node(number of node can be either 1,2,3, etc.)
  String command(info, ring, stop, etc.)
  String arguments(-- CLEANUP, -- tokesn, etc. In case no arguments needed for that command, user will  
                              have to type in "" to make it a null argument). I am not sure if there is a better way to  
                              deal with this. Let me know if there is and I can look it up.)

Also, I have moved 'Runtime runtime = Runtime.getRuntime()' to Bridge and made it protected to be accessed by CCMBridge.

### CODE REVIEW

File: test/validation/org/apache/cassandra/bridges/Bridge.java [hunk 1]
```diff
@@ -20,10 +20,44 @@
...
     public abstract void stop();
     public abstract void destroy();
     public abstract String readClusterLogs();
     public abstract void captureLogs(String testName);
+
+    public void nodeTool(int node, String command, String arguments)
+    {
+        try
+        {
+            String fullCommand = "ccm node" + node + " nodetool " + command + " " + arguments;
```
Comment (ptnapoleon) [inline test/validation/org/apache/cassandra/bridges/Bridge.java]: Why is this in Bridge, and not CCMBridge? This implementation is very ccm specific.

File: test/validation/org/apache/cassandra/bridges/Bridge.java [hunk 2]
```diff
@@ -20,10 +20,44 @@
...
+            logger.debug("Executing: " + fullCommand);
+            Process p = runtime.exec(fullCommand);
```
Comment (ptnapoleon) [inline test/validation/org/apache/cassandra/bridges/Bridge.java]: Why not route this through CCMBridge#execute or executeAnd(Read|Print)?

File: test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java [hunk 3]
```diff
@@ -35,8 +35,6 @@
 
     private int nodeCount;
     static final File CASSANDRA_DIR = new File("./");
-
-    private final Runtime runtime = Runtime.getRuntime();
```
Comment (ptnapoleon) [inline test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:39]: No objection to this change. The other bridges will most likely re-implement this, so smart to move it to the base class.

File: test/validation/org/apache/cassandra/bridges/Bridge.java (hunk 1 above)
Comment (sylvestor88) [inline test/validation/org/apache/cassandra/bridges/Bridge.java]: I thought from our previous discussion that this function needed to be implemented in Bridge as the modules will have access to Bridge class. So, should I just create an abstract function in Bridge and implement the function in CCMBridge?

File: test/validation/org/apache/cassandra/bridges/Bridge.java (hunk 2 above)
Comment (sylvestor88) [inline test/validation/org/apache/cassandra/bridges/Bridge.java]: Will do that if moving the nodeTool function to the CCMBridge.

File: test/validation/org/apache/cassandra/bridges/Bridge.java (hunk 1 above)
Comment (ptnapoleon) [inline test/validation/org/apache/cassandra/bridges/Bridge.java]: The modules will now have access to their Bridge, that is true. But, with inheritance, that means they can run any subclass of Bridge. So yes, Bridge should have an abstract function, and this implementation should be in CCMBridge.

**Comment by sylvestor88** [comment]
Pushed new changes

**Comment by ptnapoleon** [comment]
Looks great!

**Pull request merged**